### ISSUE TITLE
azurerm_data_factory_dataset_snowflake does not show table name in ADF

### ISSUE THREAD

**Issue by jamesonuk** [hashicorp/terraform-provider-azurerm#17275]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.1

### AzureRM Provider Version

3.8.0

### Affected Resource(s)/Data Source(s)

azurerm_data_factory_dataset_snowflake

### Terraform Configuration Files

```hcl
resource "azurerm_data_factory_dataset_snowflake" "snowflake_staging_dataset" {
  name                = "dataset_name" 
  data_factory_id     = var.data_factory_id
  linked_service_name = "snowflake-linked-service"
  schema_name         = "staging_schema"
  table_name          = "table_name"
  folder              = "ADF Folder"
}
```

### Debug Output/Panic Output

```shell
None
```

### Expected Behaviour

Schema / Table should show up in ADF

### Actual Behaviour

Going into ADF -> Author -> Dataset it shows the table name as loading for a second then changes to None
![image](https://user-images.githubusercontent.com/1040621/174076971-3da9f7ec-7660-47de-ab1c-76ca675d4493.png)
The table is correctly mapped and preview data works.

This generates the following JSON
```{
    "name": "FIN_AAGLANGUAGE_FND_SNOWFLAKE",
    "properties": {
        "linkedServiceName": {
            "referenceName": "snowflake-linked-service",
            "type": "LinkedServiceReference"
        },
        "folder": {
            "name": "ADF Folder"
        },
        "annotations": [],
        "type": "SnowflakeTable",
        "typeProperties": {
            "schema": "Staging name",
            "table": "table_name"
        }
    },
    "type": "Microsoft.DataFactory/factories/datasets"
}
```

Adding in `schema: []` after `annotations: []` makes the schema / table name appear in ADF

### Steps to Reproduce

terraform apply

**Comment by tombuildsstuff** [comment]
@ms-henglu FWIW this is also a Portal/ADF bug which wants raising with that team?

**Comment by tombuildsstuff** [comment]
Fixed via #17346

**Comment by ms-henglu** [comment]
@tombuildsstuff - Got it, will do. Thanks!
